frontend/src/components/DismissingComponentWrapper.js

```diff
@@ -0,0 +1,47 @@
+import React, { useEffect } from 'react';
+import PropTypes from 'prop-types';
+
+function DismissingComponentWrapper({
+  shown, timeVisibleInSec, hideFromScreenReader, children, updateShown,
+}) {
+  useEffect(() => {
+    let id;
+    if (shown) {
+      id = setTimeout(() => {
+        updateShown(false);
+      }, [timeVisibleInSec * 1000]);
+    } else {
+      clearTimeout(id);
+    }
+
+    return () => {
+      clearTimeout(id);
```

Comment:
Should this check if `id` has been set before trying to clear?

Reply:
Passing an invalid id [does nothing](https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/clearTimeout#notes). We could always check just to be safe though.

Reply:
Nah, no reason to add code to guard against the impossible.